pkg/model/multiversion/api_info.go

```diff
@@ -0,0 +1,38 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package multiversion
+
+// TODO(a-hilaly) move this file outside of pkg/generate/multiversion. Ideally we
+// Should be able to access APIStatus and APIInfo to prevent regenerating removed or
+// deprecated APIs.
+// TODO(a-hilaly) store API status in `ack-generate-metadata.yaml`
+
+type APIStatus string
+
+const (
+	APIStatusUnknown    APIStatus = "unknown"
+	APIStatusAvailable            = "available"
+	APIStatusRemoved              = "removed"
+	APIStatusDeprecated           = "deprecated"
+)
```

Comment:
Are these in line with the statuses used by AWS or K8s? I tried to do a bit of googling but didn't find anything useful. I did see `stable`, `deprecated` and `preview`.

Reply:
These are k8s api versions statuses (picked from deprecation table https://kubernetes.io/docs/reference/using-api/deprecation-policy/#deprecating-parts-of-the-api)

Reply:
I'm thinking that maybe we can split `available` to `stable` and `preview`. WDYT @RedbackThomson ?

Reply:
> I'm thinking that maybe we can split `available` to `stable` and `preview`. WDYT @RedbackThomson ?

I'm against this. Our use of the phases `preview` and `stable` have nothing to do with API versions. They are an indication of the confidence of the team that owns that controller about the long-running stability of a particular controller image.

Reply:
Alright - keeping `available`/`removed`/`deprecated` for now.